### ISSUE TITLE
Add parameters to configure log rotation

### ISSUE THREAD

**Issue by abitmore** [bitshares/bitshares-core#805]
Currently log rotation of `witness_node` is hard-coded as
* one log file per hour and 
* keep only log files created within one day.

It's better to change them to configurable for easier data analysis and trouble shooting. I personally want to keep the log files for at least one week.

Related code is [here](https://github.com/bitshares/bitshares-core/blob/test-2.0.180328/programs/witness_node/main.cpp#L363-L371).
```
            // construct a default file appender config here
            // filename will be taken from ini file, everything else hard-coded here
            fc::file_appender::config file_appender_config;
            file_appender_config.filename = file_name;
            file_appender_config.flush = true;
            file_appender_config.rotate = true;
            file_appender_config.rotation_interval = fc::hours(1);
            file_appender_config.rotation_limit = fc::days(1);
```

**Comment by cifer-lee** [comment]
Pls check this pr?
https://github.com/bitshares/bitshares-core/pull/840

**Comment by abitmore** [comment]
Merged #840. Closing.
